--- old/usr/src/cmd/fm/modules/common/zfs-retire/zfs_retire.c	Sat Apr 15 12:44:35 2017
+++ new/usr/src/cmd/fm/modules/common/zfs-retire/zfs_retire.c	Sat Apr 15 12:44:35 2017
@@ -427,6 +427,14 @@
 		    &retire) == 0 && retire == 0)
 			continue;
 
+		if (fmd_nvl_class_match(hdl, fault,
+		    "fault.io.disk.ssm-wearout") &&
+		    fmd_prop_get_int32(hdl, "ssm_wearout_skip_retire") ==
+		    FMD_B_TRUE) {
+			fmd_hdl_debug(hdl, "zfs-retire: ignoring SSM fault");
+			continue;
+		}
+
 		/*
 		 * While we subscribe to fault.fs.zfs.*, we only take action
 		 * for faults targeting a specific vdev (open failure or SERD
@@ -561,6 +569,7 @@
 
 static const fmd_prop_t fmd_props[] = {
 	{ "spare_on_remove", FMD_TYPE_BOOL, "true" },
+	{ "ssm_wearout_skip_retire", FMD_TYPE_BOOL, "true"},
 	{ NULL, 0, NULL }
 };
 
