--- old/usr/src/cmd/fm/modules/common/disk-transport/disk-transport.conf	Sat Apr 15 23:15:27 2017
+++ new/usr/src/cmd/fm/modules/common/disk-transport/disk-transport.conf	Sat Apr 15 23:15:27 2017
@@ -21,5 +21,8 @@
 #
 # Copyright 2007 Sun Microsystems, Inc.  All rights reserved.
 # Use is subject to license terms.
+# Copyright 2016 Nexenta Systems, Inc.  All rights reserved.
 #
-#ident	"%Z%%M%	%I%	%E% SMI"
+#
+# To disable transport of "high solid state media used %" faults uncomment:
+# setprop ignore-ssm-wearout true
