--- old/usr/src/lib/fm/libdiskstatus/common/ds_impl.h	Sat Apr 29 19:34:03 2017
+++ new/usr/src/lib/fm/libdiskstatus/common/ds_impl.h	Sat Apr 29 19:34:03 2017
@@ -21,13 +21,12 @@
 /*
  * Copyright 2007 Sun Microsystems, Inc.  All rights reserved.
  * Use is subject to license terms.
+ * Copyright 2016 Nexenta Systems, Inc.  All rights reserved.
  */
 
 #ifndef	_DS_IMPL_H
 #define	_DS_IMPL_H
 
-#pragma ident	"%Z%%M%	%I%	%E% SMI"
-
 #include <dlfcn.h>
 #include <libnvpair.h>
 #include <sys/types.h>
@@ -53,6 +52,7 @@
 	nvlist_t		*ds_overtemp;	/* overtemp */
 	nvlist_t		*ds_predfail;	/* predict fail */
 	nvlist_t		*ds_testfail;	/* self test fail */
+	nvlist_t		*ds_ssmwearout;	/* SSM wearout fail */
 	int			ds_error;	/* last error */
 	nvlist_t		*ds_state;	/* protocol state */
 };
@@ -60,6 +60,7 @@
 #define	DS_FAULT_OVERTEMP	0x1
 #define	DS_FAULT_PREDFAIL	0x2
 #define	DS_FAULT_TESTFAIL	0x4
+#define	DS_FAULT_SSMWEAROUT	0x8
 
 extern void dprintf(const char *, ...);
 extern void ddump(const char *, const void *, size_t);
