--- old/usr/src/uts/common/sys/fm/io/scsi.h	Sat Apr 15 12:44:36 2017
+++ new/usr/src/uts/common/sys/fm/io/scsi.h	Sat Apr 15 12:44:36 2017
@@ -21,13 +21,12 @@
 /*
  * Copyright 2007 Sun Microsystems, Inc.  All rights reserved.
  * Use is subject to license terms.
+ * Copyright 2016 Nexenta Systems, Inc.  All rights reserved.
  */
 
 #ifndef	_SYS_FM_IO_SCSI_H
 #define	_SYS_FM_IO_SCSI_H
 
-#pragma ident	"%Z%%M%	%I%	%E% SMI"
-
 #ifdef	__cplusplus
 extern "C" {
 #endif
@@ -48,6 +47,10 @@
 #define	FM_EREPORT_PAYLOAD_SCSI_CURTEMP		"current-temperature"
 #define	FM_EREPORT_PAYLOAD_SCSI_THRESHTEMP	"threshold-temperature"
 
+#define	FM_EREPORT_SCSI_SSMWEAROUT		"ssm-wearout"
+#define	FM_EREPORT_PAYLOAD_SCSI_CURSSMWEAROUT	"current-ssm-wearout"
+#define	FM_EREPORT_PAYLOAD_SCSI_THRSHSSMWEAROUT	"threshold-ssm-wearout"
+
 #define	FM_EREPORT_SCSI_TESTFAIL		"self-test-failure"
 #define	FM_EREPORT_PAYLOAD_SCSI_RESULTCODE	"result-code"
 #define	FM_EREPORT_PAYLOAD_SCSI_ADDRESS		"address"
